### PR TITLE
Move xml doc section

### LINKED ISSUE #4176: <see> xml doc topic doesn't mention the href attribute
The syntax on the topic only shows `<see cref="member"/>` but not the see href syntax:
`<see href="url">link text</see>`

edit: added topic details

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7c9f2f59-0dec-c531-b19a-a2109ea0f973
* Version Independent ID: 74c6022c-8661-216e-4f48-e5f08f432f85
* Content: [&lt;see&gt; - C# Programming Guide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/see)
* Content Source: [docs/csharp/programming-guide/xmldoc/see.md](https://github.com/dotnet/docs/blob/staging/docs/csharp/programming-guide/xmldoc/see.md)
* Product: **dotnet-csharp**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### LINKED ISSUE #7622: Document title make it hard to find
This document is specific to how references to types/methods & other constructs are identified but it's title is very generically about the XML file, making it hard to find from google/bing whatever

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 79307f9a-ba5a-b108-918b-917df121facd
* Version Independent ID: 6d011fb8-572e-cea9-4e4e-70720dd76251
* Content: [Processing the XML File (C# Programming Guide)](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/processing-the-xml-file)
* Content Source: [docs/csharp/programming-guide/xmldoc/processing-the-xml-file.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/xmldoc/processing-the-xml-file.md)
* Product: **dotnet-csharp**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### LINKED ISSUE #9355: Table examples
@LMFSoft commented on [Sat Dec 01 2018](https://github.com/dotnet/docs.it-it/issues/55)

Please add more examples for &lt;list type="table"&gt;.

---
#### Dettagli del documento

⚠ *Non modificare questa sezione. È necessaria per i collegamenti relativi ai problemi tra docs.microsoft.com ➟ GitHub.*

* ID: 40ad53e4-4786-d1a1-5f8a-2a8b914ebb37
* Version Independent ID: df94f227-fd86-0fad-e5fe-d0053d9ed321
* Content: [&lt;list&gt; (Guida per programmatori C#)](https://docs.microsoft.com/it-it/dotnet/csharp/programming-guide/xmldoc/list)
* Content Source: [docs/csharp/programming-guide/xmldoc/list.md](https://github.com/dotnet/docs.it-it/blob/live/docs/csharp/programming-guide/xmldoc/list.md)
* Product: **dotnet-csharp**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### LINKED ISSUE #15906: Single spacing
Its worth noting AND THERE SHOULD BE EXAMPLES HERE - that &lt;para&gt; started being double spaced with VS2017 and continues to be doubled spaced in VS2019.  Several issues were raised on MSDN about it and all were closed with no solution provided for creating single-spaced XML intellisense comments in our C# and F# code. All just say "not on our current road map".

It is now necessary to add &lt;br/&gt; breaks at the end of each line you want to have single spaced.
For example, if you wanted to include a single spaced block of code to your method.

```csharp
        /// <summary>Gets the angle (0-360, 0 being North) between two points. The angle is the direction from point 1 to point 2.
        /// <code> <para>
        /// GetAngle(0, 0, 1, 0) = 0<br/>
        /// GetAngle(0, 0, -1, 0) = 180<br/>
        /// GetAngle(0, 0, -1, 0) = 180<br/>
        /// GetAngle(0, 0, 0, 1) = 90<br/>
        /// GetAngle(0, 0, 0, -1) = 270<br/>
        /// </para></code>
        /// </summary>
        /// <param name="lat1"></param>
        /// <param name="long1"></param>
        /// <param name="lat2"></param>
        /// <param name="long2"></param>
        /// <returns></returns>
```

edit by @billwagner: format sample.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 48fc95e1-59ca-895f-dc67-b7397363689a
* Version Independent ID: 8cbd9296-638d-5e83-24a6-569b497835e2
* Content: [&lt;para&gt; - C# Programming Guide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/para)
* Content Source: [docs/csharp/programming-guide/xmldoc/para.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/xmldoc/para.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### LINKED ISSUE #16204: Include information on Generic types
If I want to include a reference to a generic type, the format isn't clear. The format should be of the form:
&lt;see cref="IDictionary{T, U}" /&gt;

We should document how to format this. Also (I'm unsure if possible), indicate how to reference T and U (in this example) as concrete type (when it is known at compile time what those types are).

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7c9f2f59-0dec-c531-b19a-a2109ea0f973
* Version Independent ID: 74c6022c-8661-216e-4f48-e5f08f432f85
* Content: [&lt;see&gt; - C# Programming Guide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/see)
* Content Source: [docs/csharp/programming-guide/xmldoc/see.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/xmldoc/see.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### LINKED ISSUE #17326: inheritdoc is per member?
For example with this interface:

```cs
/// <summary>
/// You may have some primary information about this interface.
/// </summary>
public interface ITestInterface
{
    /// <summary>
    /// Method documentation.
    /// </summary>
    void Method();
}

///<inheritdoc cref="ITestInterface"/>
public class TestClass : ITestInterface
{
    public void Method()
    {
    {
}
```

Method in TestClass will inherit the documentation also or do I have to add another &lt;nheritdoc /&gt; in Method as well?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 04441bbc-7e98-d6df-2c5a-20e816d79dc7
* Version Independent ID: ba155807-a2b0-8982-abe7-fac1e2b51a7d
* Content: [&lt;inheritdoc&gt; - C# Programming Guide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/inheritdoc)
* Content Source: [docs/csharp/programming-guide/xmldoc/inheritdoc.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/xmldoc/inheritdoc.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### LINKED ISSUE #17625: Syntax description missing
The example shows that "cref=" can be used. This is not written in the description itself. So I wonder whether the syntax includes other possible constructs that are not described. Please update.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 04441bbc-7e98-d6df-2c5a-20e816d79dc7
* Version Independent ID: ba155807-a2b0-8982-abe7-fac1e2b51a7d
* Content: [&lt;inheritdoc&gt; - C# Programming Guide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/inheritdoc#feedback)
* Content Source: [docs/csharp/programming-guide/xmldoc/inheritdoc.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/xmldoc/inheritdoc.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### LINKED ISSUE #17796: Escaping <>
You really should discuss all the ways that &lt;&gt;'s need to be escaped in various situations. For example, in crefs, in normal comment text, in &lt;code&gt; and &lt;c&gt; elements, etc. You examples all conveniently sidestep this constant irritant. The move to DocFX adds additional complexity since it allows markdown in comments, and markdown has its own, often conflicting escaping rules for &lt;&gt;'s.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: e6f92e76-2785-188b-6046-ec741144fcd0
* Version Independent ID: 8708e9f1-a45f-4106-6ba2-bf2e56f76aab
* Content: [&lt;example&gt; - C# programming guide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/example#feedback)
* Content Source: [docs/csharp/programming-guide/xmldoc/example.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/xmldoc/example.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### LINKED ISSUE #18236: Using the <inheritdoc /> Tag
Hello,

I want to thank you for the new updates of Visual Studio IDE and all the shiny .netCore. I think the documentation is not cathing up though and might need to be updated to reflect some basic use-case scenarios.

Comparing Microsoft's documentation link:
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/inheritdoc
to something like:
https://tunnelvisionlabs.github.io/SHFB/docs-master/SandcastleBuilder/html/79897974-ffc9-4b84-91a5-e50c66a0221d.htm
For example, it's not obvious there can be nested inheritance in the above link #Nested

or the new feature such as the quick info style support for XML comments and using
<c>code<c><em>emphasis</em><strong>strong</strong>
https://docs.microsoft.com/en-us/visualstudio/releases/2019/release-notes-v16.3#16.3.0

is not reflected in
https://docs.microsoft.com/en-us/dotnet/csharp/codedoc

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 04441bbc-7e98-d6df-2c5a-20e816d79dc7
* Version Independent ID: ba155807-a2b0-8982-abe7-fac1e2b51a7d
* Content: [&lt;inheritdoc&gt; - C# Programming Guide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/inheritdoc)
* Content Source: [docs/csharp/programming-guide/xmldoc/inheritdoc.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/xmldoc/inheritdoc.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### LINKED ISSUE #18659: Comment Tags missing
There are more tags than described here and in the spec.
Please add the tags, which are not yet documented, into the spec and and to the documentation tag reference (a sub-page of this).
So far I found: &lt;e&gt;, &lt;i&gt;

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 306c4f25-192e-0980-c663-d829f38ef7c4
* Version Independent ID: 8918a37b-cbfa-33f1-21e2-6bf5ea3ccb54
* Content: [XML documentation comments - C# programming guide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/)
* Content Source: [docs/csharp/programming-guide/xmldoc/index.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/xmldoc/index.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### LINKED ISSUE #18730: Indicate which tags show in intellisense
I can find no list of which tags are formatted into intellisense help and how. e.g. Through trial and error, I know <remarks> is a useless tag because it doesn't go into intellisense. It would be nice to have that documented somewhere reachable from this page.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7d5aa865-b9d2-e977-a272-3886aa0a1111
* Version Independent ID: 353b1c44-b90a-e5cc-a193-7b42ec714290
* Content: [Recommended tags for documentation comments - C# programming guide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/recommended-tags-for-documentation-comments)
* Content Source: [docs/csharp/programming-guide/xmldoc/recommended-tags-for-documentation-comments.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/xmldoc/recommended-tags-for-documentation-comments.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### LINKED ISSUE #18818: Clean up xmldoc articles
For example, `<seealso>` must be outside the `<summary>` tags.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 337464b6-c9f4-87ee-f774-5f4509c3830e
* Version Independent ID: 9626e965-109d-5e3c-eb42-ef8fbccf2fb4
* Content: [&lt;seealso&gt; - C# programming guide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/seealso)
* Content Source: [docs/csharp/programming-guide/xmldoc/seealso.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/xmldoc/seealso.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### LINKED ISSUE #21885: <see> in C# Guide doesn't mention inner text, as shown in F# guide
In [Document your code with XML comments](https://docs.microsoft.com/en-us/dotnet/fsharp/language-reference/xml-documentation#recommended-tags) in the F# Guide, `<see>` is documented as `<see cref="reference">text</see>` where "The *text* is the text shown in the link."

[`<see>` in the C# guide](https://docs.microsoft.com/dotnet/csharp/programming-guide/xmldoc/see) does not document whether inner text is allowed or what behavior it implies.

Is `<see>` inner text permitted in both C# and F#?  (I tested that `<see>` inner text is preserved in the XML documentation file generated from C#.)  In either case, perhaps the C# Guide should be updated to mention it.

Thanks for considering,
Kevin

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7c9f2f59-0dec-c531-b19a-a2109ea0f973
* Version Independent ID: 74c6022c-8661-216e-4f48-e5f08f432f85
* Content: [&lt;see&gt; - C# programming guide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/see)
* Content Source: [docs/csharp/programming-guide/xmldoc/see.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/xmldoc/see.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### LINKED ISSUE #23533: Path attribute on <inheritdoc/>
The ```path``` attribute is missing.

I only wanted to include the "returns" part of a inherited documentation.
Had to waste time looking up that ```path``` attribute exists, XPath expression is a thing, and had to learn how to actually do it.

Noting that ```path``` attribute is a XPath expression, with references/examples to what can be done in it would be great.

Here's the paths I use; could be used as examples I guess:
• Inherit only the ```returns``` tag: ```path="/returns"```
• Inherit all but ```remarks``` tags: ```path="//*[not(self::remarks)]"```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 04441bbc-7e98-d6df-2c5a-20e816d79dc7
* Version Independent ID: ba155807-a2b0-8982-abe7-fac1e2b51a7d
* Content: [&lt;inheritdoc&gt; - C# Programming Guide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/xmldoc/inheritdoc)
* Content Source: [docs/csharp/programming-guide/xmldoc/inheritdoc.md](https://github.com/dotnet/docs/blob/main/docs/csharp/programming-guide/xmldoc/inheritdoc.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by BillWagner**
Rework the XML documentation section and move it into the language reference.

Notes for reviewers:

You should view the three new files under language-reference/xmldoc as all new content. They were taken from existing content, but substantially rearranged and rewritten. The samples and snippets were also consolidated and moved. The final code has a smaller set of samples, but each sample represents a typical scenario: documenting a class or record, inheriting docs from other locations, and generic classes and methods.

- The older content was removed and redirects added to the new files.
- A lengthy list of issues regarding XML comments were all addressed. (see list below)
- Finally, clean repo was run to replace redirects and replace absolute links.

Fixes #4176: Examples and recommended content now include an `href` example for `see` and `seealso`.
Fixes #7622:  This content is in the index for this section, which increases its findability. The title has also changed.
Fixes #9355: The examples now include a table for the list tag.
Fixes #15906: Note that `para` creates a double spaced paragraph. Use `<br/>` for single spacing.
Fixes #16204: The `cref` tag explains how to reference a generic type. The generic examples include an example.
Fixes #17326: Specify that `inheritdoc` for a type also inherits comments for all members.
Fixes #17625: No actual change. Verify that `cref` and `Path` are the only tags to specify the inherited source.
Fixes #17796:  The introduction to "recommended tags" explains how to escape `<` and `>` in comments.
Fixes #18236: This is a duplicate of 18730, and 23533.
Fixes #18659:  The "recommended tags" article now lists the relevant tags validated by the compiler.
Fixes #18730:  The "recommended tags" article lists which tags are shown in intellisense.
Fixes #18818: Note that some tags cannot be nested, specifically the `seealso` tag.
Fixes #21885: point out that the `<see>` tag can provide alternative text.
Fixes #23533:  Show examples of `<inheritdoc>` that use the path attribute to include only some tags.

[Internal review site link](https://review.docs.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/?branch=pr-en-us-24705) See the entire section. This is the index.

### CODE REVIEW

File: docs/csharp/language-reference/xmldoc/recommended-tags.md [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+
+### \<typeparam>
+
+```xml
+<typeparam name="TResult">The type returned from this method</typeparam>
+```
+
+- `TResult`: The name of the type parameter. Enclose the name in double quotation marks (" ").
+
+The `<typeparam>` tag should be used in the comment for a generic type or method declaration to describe a type parameter. Add a tag for each type parameter of the generic type or method. The text for the `<typeparam>` tag will be displayed in IntelliSense, the [Object Browser Window](/visualstudio/ide/viewing-the-structure-of-code#BKMK_ObjectBrowser) code comment web report.
```
Comment (gewarren) [inline docs/csharp/language-reference/xmldoc/recommended-tags.md:369]: The last sentence of this paragraph doesn't make sense to me.

File: docs/csharp/misc/cs1570.md [hunk 2]
```diff
@@ -13,13 +13,13 @@ XML comment on 'construct' has badly formed XML — 'reason'
 
 When using [**DocumentationFile**](../language-reference/compiler-options/output.md#documentationfile), any comments in the source code must be in XML. Any error with your XML markup will generate CS1570. For example:
 
-- If you are passing a string to a **cref**, such as in an [\<exception>](../programming-guide/xmldoc/exception.md) tag, the string must be enclosed in double quotation marks.
+- If you are passing a string to a **cref**, such as in an [\<exception>](../language-reference/xmldoc/recommended-tags.md) tag, the string must be enclosed in double quotation marks.
 
-- If you are using a tag, such as [\<seealso>](../programming-guide/xmldoc/seealso.md), which does not have a closing tag, you must specify a forward slash before the closing angle bracket.
+- If you are using a tag, such as [\<seealso>](../language-reference/xmldoc/recommended-tags.md), which does not have a closing tag, you must specify a forward slash before the closing angle bracket.
```
Comment (gewarren) [inline docs/csharp/misc/cs1570.md:18]:
```suggestion
- If you're using a tag that doesn't have a closing tag, such as [\<seealso>](../language-reference/xmldoc/recommended-tags.md), you must specify a forward slash before the closing angle bracket.
```

**Review by gewarren — COMMENTED** [review]
Looking great!

**Review by tdykstra — APPROVED** [review]
This is great work, a tremendous amount of text and code and all done extremely well.  I noted some typos and some opportunities to improve readability - feel free to take or leave any of the suggestions and comments.

File: docs/csharp/language-reference/xmldoc/index.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+ms.assetid: 803b7f7b-7428-4725-b5db-9a6cff273199
+---
+# XML documentation comments
+
+C# source files can have structured comments that produce API documentation for the types defined in those files. The C# compiler produces an *XML* file that contains structured data representing the comments and the API signatures. Other tools can process that XML output to create human-readable documentation. Formats include web pages, PDF files, and other formats.
+
+This process provides many advantages for you to add API documentation in your code:
+
+- The C# compiler combines the structure of the C# code with the text of the comments into a single XML document.
+- The C# compiler validates the comments match the API signatures for relevant tags.
```
[the remark this answers is not in the document]
Comment (BillWagner) [reply docs/csharp/language-reference/xmldoc/index.md]: I think I'm going to leave the example off here. I've added more where the tags, such as `<param>` are discussed.

File: docs/csharp/language-reference/xmldoc/examples.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+## Document a class, struct, or interface
+
+The following example shows common language elements, and the tags you'll likely use to describe these elements.  The documentation comments describe the use of the tags, rather than the class itself.
+
+:::code language="csharp" source="./snippets/xmldoc/DocComments.cs" ID="ClassExample":::
+
+## Document a hierarchy of classes and interfaces
+
+The `<InheritDoc>` element means a type or member *inherits* documentation comments from a base class or interface. You can also use the `<InheritDoc>` element with the `cref` attribute to inherit comments from a member of the same type. The following example shows ways to use this tag. Note that when you add the `inheritdoc` attribute to a type, all member comments are also inherited.
```
[the remark this answers is not in the document]
Comment (BillWagner) [reply docs/csharp/language-reference/xmldoc/examples.md:19]: both work, but I standardized on `inheritdoc`

File: docs/csharp/language-reference/xmldoc/examples.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+:::code language="csharp" source="./snippets/xmldoc/DocComments.cs" ID="IncludeTag":::
+
+The second file, *xml_include_tag.xml*, contains the documentation comments.
+
+:::code language="xml" source="./snippets/xmldoc/xml_include_tag.xml" :::
+
+## Generic types
+
+Generic types and methods will use the `<TypeParam>` tag to describe type parameters. Also, the value for the `cref` attribute requires new syntax to reference a generic method or class:
```
[the remark this answers is not in the document]
Comment (BillWagner) [reply docs/csharp/language-reference/xmldoc/examples.md]: both work, I standardized on "typeparam"

File: docs/csharp/language-reference/xmldoc/index.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+    - return type
+    - ELEMENT_TYPE_SENTINEL
+
+The following examples show how the ID strings for a class and its members are generated:
+
+:::code language="csharp" source="./snippets/xmldoc/idstrings.cs":::
+
+## C# language specification
+
+[!INCLUDE[CSharplangspec](~/includes/csharplangspec-md.md)]
```
[the remark this answers is not in the document]
Comment (BillWagner) [reply docs/csharp/language-reference/xmldoc/index.md:162]: Yup. I updated the link to the documentation comments section.

File: docs/csharp/language-reference/xmldoc/index.md [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+- [Doxygen](https://github.com/doxygen/doxygen): *Doxygen* generates an on-line documentation browser (in HTML) or an off-line reference manual (in LaTeX) from a set of documented source files. There's also support for generating output in RTF (MS-Word), PostScript, hyperlinked PDF, compressed HTML, DocBook, and Unix man pages. You can configure Doxygen to extract the code structure from undocumented source files.
+
+### ID strings
+
+The compiler generates an ID string for each construct in your code that's tagged to generate documentation. (For information about how to tag your code, see [Recommended Tags for Documentation Comments](./recommended-tags.md).) The ID string uniquely identifies the construct. Programs that process the XML file use the ID string to identify the corresponding .NET metadata or reflection item that the documentation applies to.
```
[the remark this answers is not in the document]
Comment (BillWagner) [reply docs/csharp/language-reference/xmldoc/index.md:117]: I re-wrote this opening. Great suggestion.

**Pull request merged**